autoemulate/experimental/save.py

```diff
@@ -0,0 +1,156 @@
+
+        # Save metadata to CSV
+        metadata_path = Path(f"{full_path}_metadata.csv")
+        metadata_df = result.metadata_df()
+        try:
+            metadata_df.to_csv(metadata_path, index=False)
+            self.logger.info("Metadata saved to %s", metadata_path)
+        except Exception as e:
+            self.logger.error("Failed to save metadata to %s: %s", metadata_path, e)
+            raise
+
+        return full_model_path
+
+    def _load_model(self, path: str | Path):
+        """Loads a model from disk.
+
+        Parameters
+        ----------
+        path : str or Path
+            Path to load the model.
+        """
+        path = Path(path)
+        try:
+            model = joblib.load(path)
+            self.logger.info("Model loaded from %s", path)
+            return model
+        except Exception as e:
+            self.logger.error("Failed to load model from %s: %s", path, e)
+            raise
+
+    def _load_result(self, path: str | Path) -> Result | Emulator:
+        """Loads a model and (if it exists) its metadata from disk,
+        returning either a Result or Emulator object.
+
+        Parameters
+        ----------
+        path : str or Path
+            Path to the model file.
+
+        Returns
+        -------
+        Result or Emulator
+            The loaded model or result object.
+        """
+        model = self._load_model(path)
+        metadata_path = Path(f"{path}_metadata.csv")
+        try:
+            metadata_df = pd.read_csv(metadata_path, nrows=1)
+            self.logger.info("Metadata loaded from %s", metadata_path)
+        except Exception as e:
+            msg = "Failed to load metadata from %s: %s"
+            self.logger.error(msg, metadata_path, e)
+            return model
+        row = metadata_df.iloc[0]
+        config = row["config"]
+        config = ast.literal_eval(config)
+
+        return Result(
+            id=row["id"],
+            model_name=row["model_name"],
+            model=model,
+            config=config,
+            r2_test=row["r2_test"],
+            rmse_test=row["rmse_test"],
+            r2_test_std=row["r2_test_std"],
+            rmse_test_std=row["rmse_test_std"],
+            r2_train=row["r2_train"],
+            rmse_train=row["rmse_train"],
+            r2_train_std=row["r2_train_std"],
+            rmse_train_std=row["rmse_train_std"],
+        )
+
+    def _prepare_path(self, path, model_name):
+        """Prepares path for saving model."""
+        if path is None:
+            full_path = Path(model_name)
+        else:
+            full_path = Path(path)
+            if full_path.is_dir():
+                full_path = full_path / model_name
+
+        full_path.parent.mkdir(parents=True, exist_ok=True)
+        return full_path
+
+    @staticmethod
+    def _get_model_name(model):
+        """Gets the name of the model to save."""
+        model_name = model.model_name()
+        return f"{model_name}.joblib"
```

Comment:
Related to comment above https://github.com/alan-turing-institute/autoemulate/pull/629/files#r2225105700 this seems to add the extension?